### ISSUE TITLE
Gideon, Champion of Justice immediately sent to grave with +0 ability

### ISSUE THREAD

**Issue by drmDev** [magefree/mage#2069]
Gatecrash's Gideon, Champion of Justice is sent to grave right after you activate his +0 ability to make him a creature with P/T equal to loyalty.

Looking at the code he originally starts as a token with P/T 0/0 then there is a static ability added to set his power and toughness to his loyalty. The SetPowerToughness part does not seem to be functioning as it should. For instance, if I make his initial P/T 1/1 instead of 0/0 Gideon at least remains on the battlefield, but then only deals 1 damage when attacking. Despite only dealing 1 damage in this case, in the bottom-right of the card you can see a glimpse of his P/T and it appears to be set correctly to his loyalty (was 5 when I was testing it, +1'd him the first turn he was on battlefield).

Haven't had enough time to explore it further. Apparently this was a bug for some time - at least since the previous release, perhaps further? Not sure when it started.

**Comment by drmDev** [comment]
The change I made was to BecomeCreatureSourceEffect where I set the CharacterDefining bool to true. I noticed a similar card to how Gideon's ability is handled in CMD: "Svogthos, the Restless Tomb" that did this. However, I am informed Gideon doesn't have a CDA. So my way of resolving this issue is likely not correct.

Nonetheless, the code change here does make him work as intended in the most basic sense. He becomes a creature with indestructible and P/T equal to loyalty until end of turn. He deals damage equal to his loyalty as normal. He loses all those abilities at end of turn.
